exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -280,12 +280,10 @@ protected void validateFrom(
             changeNamesIfTableIsTemporary(tempNode);
 
             // Check the schema and throw a valid SchemaNotFound exception instead of TableNotFound exception.
-            if (catalogReader.getTable(tempNode.names) == null) {
-              catalogReader.isValidSchema(tempNode.names);
-            }
+            catalogReader.isValidSchema(tempNode.names);
```

Comment:
Names cannot be null?

Reply:
Good question, when it is set in the constructor or using a setter of `SqlIdentifier` class, it cannot be null, since there is used `ImmutableList.copyOf()` method for incoming lists. But this field is public and may be changed in some other places (I didn't find any place where it is set to null in the current Drill and Calcite code).

Since Calcite's `SqlIdentifier` code does not assume that `names` may be null (there is a lot of code where this field is used without checks), I think we should also do not check it.